salt/_modules/metalk8s_cordon.py

```diff
@@ -4,80 +4,22 @@
 This module's functions are merged into the `metalk8s_kubernetes`
 module when called by salt by virtue of its `__virtualname__` attribute.
 '''
-import logging
-
-from salt.exceptions import CommandExecutionError
-
-try:
-    import kubernetes
-    import kubernetes.client
-    from kubernetes.client.rest import ApiException
-    from urllib3.exceptions import HTTPError
-
-    HAS_LIBS = True
-except ImportError:
-    HAS_LIBS = False
-
-log = logging.getLogger(__name__)
 
 __virtualname__ = 'metalk8s_kubernetes'
 
 
 def __virtual__():
-    '''
-    Check dependencies
-    '''
-    if HAS_LIBS:
-        return __virtualname__
-
-    return False, 'python kubernetes library not found'
+    return __virtualname__
```

Comment:
You don't need the `kubernetes.client` dependency, sure, but you still need some methods from another `metalk8s_kubernetes` module. I think we should do something like:

```
def __virtual__():
    if 'metalk8s_kubernetes.update_object' not in __salt__:
        return False, 'Missing dependency in "metalk8s_kubernetes" module.'
    return __virtualname__
```

Maybe it's subject to ordering issues though (if Salt tries to load this `__virtual__` before the other one in `metalk8s_kubernetes.py`...), I'm not entirely clear on this.

Reply:
> Maybe it's subject to ordering issues though (if Salt tries to load this `__virtual__` before the other one in `metalk8s_kubernetes.py`...), I'm not entirely clear on this.

Yes maybe that why I didn't add any check for the moment, TBC

Reply:
Just checked we cannot use `__salt__` in `__virtual__` in module 

> Since __virtual__ is called before the module is loaded, __salt__ will be unreliable as not all modules will be available at this point in time. 

https://docs.saltstack.com/en/latest/ref/modules/#virtual-function

Reply:
Makes sense. Then I guess, we would need to somehow catch a potential `KeyError` when accessing `__salt__['metalk8s_kubernetes.update_object']` at runtime.